operator/pkg/utils/imageprovider/imageprovider.go

```diff
@@ -60,9 +60,9 @@ func CoreDNS() string {
 }
 
 func AWSIamAuthenticator() string {
```

Comment:
I'm not sure how I feel about this image provider abstraction. It breaks the encapsulation of the components themselves (e.g. the actual authenticator), for seemingly little value.

Reply:
The reason I added this package was to have all the images defined at a common place and not spread them all over the code. So if we need to change a tag/version for a component we can just change in this one file.

Reply:
Willing to see how this goes.